crypto/rsa/rsa_lib.c

```diff
@@ -402,6 +402,31 @@ int RSA_get0_multi_prime_crt_params(const RSA *r, const BIGNUM *exps[],
     return 1;
 }
 
+const BIGNUM *RSA_get0_n(const RSA *r)
+{
+    return r->n;
+}
+
+const BIGNUM *RSA_get0_e(const RSA *r)
+{
+    return r->e;
+}
+
+const BIGNUM *RSA_get0_d(const RSA *r)
+{
+    return r->d;
+}
+
+const BIGNUM *RSA_get0_p(const RSA *r)
+{
+    return r->p;
+}
+
+const BIGNUM *RSA_get0_q(const RSA *r)
+{
+    return r->q;
+}
+
```

Comment:
What about the CRT params?

Reply:
In this case I was not sure whether there would ever be any need to retrieve one of the dmp1, dmq1, and iqmp members separately, because they only make sense together. 	That's why I was [seeking your feedback](https://github.com/openssl/openssl/pull/6290#issuecomment-390121868) first. If you prefer having them, I will add them of course. (Just raise the thumb.)

Reply:
That was my line of thinking with the other "awkward" getters as well, so I'm thinking that if we're making separate (and admitedly more convenient) getters for the other numbers, I see no real reason to leave the CRT factors out.

Reply:
Sorry that I missed your earlier request for feedback...

Reply:
Engines implementing RSA may have a need to get the CRT parameters.
An example is the QAT Engine here:
https://github.com/intel/QAT_Engine/blob/master/qat_rsa.c#L381

It would be more consistent to add similar getters for CRT factors and key parts or not at all. It seems more confusing to add one without the other but I don't feel strongly about this.

Reply:
So getters for key parts and one gettor for *all* the CRT parts?  (NULL meaning don't return that?)

Reply:
I concur that without a use case where you want the individual CRT parts it shouldn't be added.
I guess the same is true for:
ECDSA_SIG_get0()
which hasn't been considered in this pull request.
I don't think you would want to retrieve r or s on their own so there is no point in adding simplified getters.

Reply:
> That was my line of thinking with the other "awkward" getters as well, so I'm thinking that if we're making separate (and admitedly more convenient) getters for the other numbers, I see no real reason to leave the CRT factors out.

I'm going to follow @levitte's advice. Since I've now started to add functional-style getters it does not make sense to make sense to draw an arbitrary line between getters that might be needed and those that might not, since such a judgement will always be subjectively perceived.

I've split the commits thematically and also added the getters for ECDSA_SIG as suggested by @stevelinsell.